File: features/document/src/main/java/com/nivasafinance/features/document/exception/DocumentNotFoundException.java
package com.nivasafinance.features.document.exception;

import com.nivasafinance.common.exception.ResourceNotFoundException;
import org.springframework.context.MessageSource;
import org.springframework.context.i18n.LocaleContextHolder;

import java.io.Serial;
import java.util.Locale;
import java.util.UUID;

public class DocumentNotFoundException extends ResourceNotFoundException {
    
    @Serial
    private static final long serialVersionUID = 1L;
    
    private static final Locale locale = LocaleContextHolder.getLocale();
    public static final String DOCUMENT_NOT_FOUND_KEY = "error.document.not.found";
    
    public DocumentNotFoundException(String message) {
        super(message);
    }
    
    public DocumentNotFoundException(UUID documentId, MessageSource messageSource) {
        this(messageSource.getMessage(DOCUMENT_NOT_FOUND_KEY, new Object[]{documentId}, locale));
    }
}

File: features/document/src/main/java/com/nivasafinance/features/document/exception/DocumentNotFoundException.kt
package com.nivasafinance.features.document.exception

import exception.ResourceNotFoundException
import org.springframework.context.MessageSource
import org.springframework.context.i18n.LocaleContextHolder
import java.util.Locale
import java.util.UUID

class DocumentNotFoundException(message: String) : ResourceNotFoundException(message) {
    companion object {
        private val locale: Locale = LocaleContextHolder.getLocale()
        const val DOCUMENT_NOT_FOUND_KEY = "error.document.not.found"
    }

    constructor(documentId: UUID, messageSource: MessageSource) : this(
        messageSource.getMessage(DOCUMENT_NOT_FOUND_KEY, arrayOf(documentId), locale)
    )
}
